### PR TITLE
tuf: add starter TUF client

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

## Summary
* Introduces a root provider interface that can supply a trusted root. the interface returned will be ported from the sigstore/protobuf-specs (TBA)
* Adds a TUF client that can satisfy the root provider. This TUF client is fairly basic, and I will enhance it in future follow-ups, including adding default options for the current public instance and staging, and other customizeable options (concurrency, better caching, etc)

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#
+# Copyright 2021 The Sigstore Authors.
```
Comment (cpanato) [inline .github/workflows/test.yml:2]: Nit: 2022

File: .github/workflows/verify.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#
+# Copyright 2021 The Sigstore Authors.
```
Comment (cpanato) [inline .github/workflows/verify.yml]: Nit: 2022

File: pkg/root/tuf/client_test.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+	testCases := []struct {
+		name          string
+		tufOpts       *TUFClientOptions
+		repoOpts      *RepositoryOptions
+		wantClientErr bool
+		wantInitErr   bool
+	}{
+		{
+			name: "fail: unkown cache type",
```
Comment (lukehinds) [inline pkg/root/tuf/client_test.go:53]: unkown / unknown

**Review by lukehinds — APPROVED** [review]
lgtm, just a spelling nit

**Review by cpanato — APPROVED** [review]
will fix the nit in a follow up

**Pull request merged**